src/lib/pubkey/pcurves/pcurves_util.h

```diff
@@ -0,0 +1,188 @@
+/*
+* (C) 2024 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_PCURVES_UTIL_H_
+#define BOTAN_PCURVES_UTIL_H_
```

Comment:
Main problem is for some technical reason I'm not bothering to learn the details of, C++ can't deduce the conversion from a statically sized `array` to a statically sized `span`. https://stackoverflow.com/questions/70983595

This can be worked around with some additional noise but the additional flexibility doesn't buy us anything in this context so I'd rather keep things as simple as possible.